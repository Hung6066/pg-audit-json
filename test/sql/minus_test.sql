CREATE EXTENSION "pg-audit-json";

-- Remove one key
SELECT '{"a": 1, "b": 2, "c": 3}'::jsonb - '{a}'::text[];

-- Remove multiple keys
SELECT '{"a": 1, "b": 2, "c": 3}'::jsonb - '{a,c}'::text[];

-- Test usage scenario where NEW - OLD

-- Removed c
SELECT '{"a": 1, "b": 2}'::jsonb
     - '{"a": 1, "b": 2, "c": 3}'::jsonb;

-- Changed b
SELECT '{"a": 1, "b": 99, "c": 3}'::jsonb
     - '{"a": 1, "b": 2, "c": 3}'::jsonb;

-- Added d
SELECT '{"a": 1, "b": 2, "c": 3, "d": 4}'::jsonb
     - '{"a": 1, "b": 2, "c": 3}'::jsonb;

-- Recursive changed b->c
SELECT '{"a":1, "b":{"c":321, "d":"test"}}'::jsonb
     - '{"a":1, "b":{"c":123, "d":"test"}}'::jsonb;

DROP EXTENSION "pg-audit-json";
