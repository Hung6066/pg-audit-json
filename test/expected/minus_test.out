CREATE EXTENSION "pg-audit-json";
-- Remove one key
SELECT '{"a": 1, "b": 2, "c": 3}'::jsonb - '{a}'::text[];
     ?column?     
------------------
 {"b": 2, "c": 3}
(1 row)

-- Remove multiple keys
SELECT '{"a": 1, "b": 2, "c": 3}'::jsonb - '{a,c}'::text[];
 ?column? 
----------
 {"b": 2}
(1 row)

-- Test usage scenario where NEW - OLD
-- Removed c
SELECT '{"a": 1, "b": 2}'::jsonb
     - '{"a": 1, "b": 2, "c": 3}'::jsonb;
 ?column? 
----------
 {}
(1 row)

-- Changed b
SELECT '{"a": 1, "b": 99, "c": 3}'::jsonb
     - '{"a": 1, "b": 2, "c": 3}'::jsonb;
 ?column?  
-----------
 {"b": 99}
(1 row)

-- Added d
SELECT '{"a": 1, "b": 2, "c": 3, "d": 4}'::jsonb
     - '{"a": 1, "b": 2, "c": 3}'::jsonb;
 ?column? 
----------
 {"d": 4}
(1 row)

-- Recursive changed b->c
SELECT '{"a":1, "b":{"c":321, "d":"test"}}'::jsonb
     - '{"a":1, "b":{"c":123, "d":"test"}}'::jsonb;
     ?column?      
-------------------
 {"b": {"c": 321}}
(1 row)

DROP EXTENSION "pg-audit-json";
